src/main/java/core/basesyntax/fruittransact/FruitTransaction.java

```diff
@@ -0,0 +1,5 @@
+package core.basesyntax.fruittransact;
+
+public interface FruitTransaction {
+    void transact(String name, int amount);
+}

```

Comment:
FruitTransaction is a DTO-like model, not an interface. This model has three parameters - fruit, amount, type. Your best bet would be to follow an implementation outlined in the project description, however, you are free to do it your own way, but your current implementation can be confusing.
What you are trying to achieve here is usually called a handler (here - `OperationHandler`)- with the method `handle`.